csrc/device_lower/pass/rng.cpp

```diff
@@ -0,0 +1,208 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <device_lower/pass/magic_zero.h>
+
+#include <device_lower/analysis/index_compute.h>
+#include <device_lower/lower2device.h>
+#include <dispatch.h>
+#include <instrumentation.h>
+#include <ir/utils.h>
+#include <kernel_ir_dispatch.h>
+
+namespace nvfuser {
+
+namespace {
+
+std::tuple<Val*, Expr*> createAndAllocNS(
+    std::string name,
+    DataType dtype = DataType::Index) {
+  Val* val = IrBuilder::create<NamedScalar>(name, dtype);
+  auto alloc = IrBuilder::create<kir::Allocate>(
+      val, MemoryType::Local, GpuLower::current()->kernel()->oneVal());
+  return std::make_tuple(val, alloc);
+}
+
+class RNGInserter : public kir::ExprMutator {
+ public:
+  using kir::ExprMutator::handle;
+  static std::vector<Expr*> insert(const std::vector<Expr*>& exprs) {
+    RNGInserter inserter(exprs);
+    return inserter.exprs_;
+  }
+
+ private:
+  Val* rng_subseq = nullptr;
```

Comment:
nit: underscore in the end `rng_subseq_`
https://google.github.io/styleguide/cppguide.html#Variable_Names